### ISSUE TITLE
Code splitting using `defineAsyncComponent` not working.

### ISSUE THREAD

**Issue by Akash187** [antfu/unplugin-vue-components#206]
Recently I came across a bug. I just realize that the code splitting in vue3 using **defineAsyncComponent** is not working.

Link to sample project to reproduce the issue [vue3-vite-lazy-loading](https://github.com/Akash187/vue3-vite-lazy-loading)

I have attached image of `build` command. With and without **unplugin-vue-components**.
<img width="1440" alt="Screenshot 2021-11-09 at 10 07 23 AM" src="https://user-images.githubusercontent.com/21134455/140864439-213f3a1b-8cb9-40b5-a3f7-78187fb36556.png">
<img width="1440" alt="Screenshot 2021-11-09 at 10 10 40 AM" src="https://user-images.githubusercontent.com/21134455/140864466-a8d6d15f-14dd-4149-9b16-454e0bc449f3.png">

**Comment by Akash187** [comment]
@antfu can you please take a look. 
One work around I just found is exclude the File in **vite.config.js**

```
import { defineConfig } from 'vite'
import vue from '@vitejs/plugin-vue'
import Components from 'unplugin-vue-components/vite'
import path from 'path'

// https://vitejs.dev/config/
export default defineConfig({
	plugins: [
		vue(),
		Components({
			exclude: [
				/\App.vue/,
				/[\\/]node_modules[\\/]/,
				/[\\/]\.git[\\/]/,
				/[\\/]\.nuxt[\\/]/
			]
		})
	]
})

```

**Comment by antfu** [comment]
Wondering why don't you use `<script setup>`? If do you so, it would just work. For options API, I think it would be a little bit tricker to exclude the manual registered components.

**Comment by Akash187** [comment]
Yeah, I was just being lazy avoided using <script setup>. Will start using it from the next project.

**Comment by nag5000** [comment]
Confirm, with `<script setup>` it just works.

With Options API auto synchronous registration (by unplugin-vue-components) overlaps manual asynchronous. There is also a warning about it during `npm run build`:
`
[plugin:vite:reporter]
(!) SecretImage.vue is dynamically imported by App.vue but also statically imported by App.vue, dynamic import will not move module into another chunk.
`

For Options API the workaround is to register a component with a different name than the original one, which is not going to be autoresolved in a template:
```vue
<template>
  <SecretImageAsync />
</template>

<script>
const SecretImageAsync = defineAsyncComponent(() => import('./components/SecretImage.vue'))

export default {
  components: {
    SecretImageAsync,
  },
}
</script>
```
